src/org/ods/services/OpenShiftService.groovy

```diff
@@ -171,4 +173,67 @@ class OpenShiftService {
       label: "Get phase of build ${buildId}"
     ).trim().toLowerCase()
   }
+  
+  public String getContainerForImage (String projectId, String rc, String image) {
+    script.sh(
+      script: """oc -n ${projectId} get rc ${rc} -o jsonpath='{.spec.template.spec.containers[?(contains .image "${image}")].name}'""",
+      returnStdout: true,
+      label: "Getting containers for ${rc} and image ${image}"
+    )
+  }
+  
+  // Gets pod of deployment
+  Map getPodDataForDeployment(String componentId, String rc) {
+    def index = 5
+    while (index > 0)
+    {
+      def podStatus = script.sh(
+        script: "oc -n ${project} get pod -l deployment=${rc} -o jsonpath='{.items[*].status.phase}'",
+        returnStdout: true,
+        label: "Getting OpenShift pod data for deployment ${rc}"
+      )
+      if (podStatus && podStatus == "Running") {
+        break
+      } else {
+        script.sleep(5000)
+        index--
+      }
+    }
+      
+    def stdout = script.sh(
+      script: "oc -n ${project} get pod -l deployment=${rc} -o json",
+      returnStdout: true,
+      label: "Getting OpenShift pod data for deployment ${rc}"
+    ).trim()
+
+    extractPodData(stdout, "deployment '${rc}'")
+  }
+
+  private Map extractPodData(String ocOutput, String description) {
+    def j = new JsonSlurperClassic().parseText(ocOutput)
+    if (j?.items[0]?.status?.phase?.toLowerCase() != 'running') {
+      throw new RuntimeException("Error: no pod for ${description} running / found.")
+    }
+
+    def podOCData = j.items[0]
+
+    // strip all data not needed out
+    def pod = [ : ]
+      pod.podName = podOCData?.metadata?.name?: "N/A"
+      pod.podNamespace = podOCData?.metadata?.namespace?: "N/A"
+      pod.podMetaDataCreationTimestamp = podOCData?.metadata?.creationTimestamp?: "N/A"
+      pod.deploymentConfigCreationEnvironment = podOCData?.metadata?.labels?.env?: "N/A"
```

Comment:
As mentioned in the other PR, I suggest to remove this.

Reply:
@michaelsauter  -. what exactly? ...

Reply:
removed .env